app/models/schedule.rb

```diff
@@ -2,4 +2,26 @@ class Schedule < ApplicationRecord
   belongs_to :schedule_run
 
   validates :case_num, :week, :hmc_score, :schedule_run_id, presence: true
+
+  def choice
+    proposal = Proposal.find_by(code: self.proposal)
+    return '' if proposal.blank? || proposal.assigned_date.nil?
```

Comment:
I think there is a misunderstanding of what the `assigned_date` field is for. This field is used by the review committee if they decide that a proposal must be scheduled on a particular date. If that happens, staff will update the assigned_date field (on the [Proposal#edit](https://pstaging.birs.ca/submitted_proposals/178/edit) page) _before_ the schedule optimizer runs.

If the `assigned_date` field is used, then [HungarianMonteCarlo](https://github.com/birs-math/proposals/blob/2193914e1638ddf2a695b89aaaf616011e44d12d/app/services/hungarian_monte_carlo.rb#L89) will set the priority to 1, and the date list to only that date, so that the HMC optimizer will always give the assigned date to that proposal.

It makes no sense to return '' as a choice if the assigned_date field is nil. The assigned date is only used by the schedule optimization program, so should not play any role in the choices column.

The choices column needs to compare the date in the Weeks column with the proposal's preferred dates.